torch/distributed/tensor/_ops/_tensor_ops.py

```diff
@@ -93,44 +93,59 @@ def propagate_single_input_strategy(op_schema: OpSchema) -> StrategyType:
 
 
 @register_op_strategy(aten.copy_.default)
-def copy_strategy(op_schema: OpSchema) -> StrategyType:
-    # TODO: this strategy is incorrect for copy_ in the case that src tensor
-    # is smaller rank than self tensor.  It is possible to select a strategy from self tensor
-    # that is invalid for dst tensor.
-    # It is also problematic to assume that shard(0) on src maps to shard(0) on self, since we
-    # may broadcast a new dim to the left or right of 0 when copying.
+def copy_inplace_strategy(op_schema: OpSchema) -> StrategyType:
+    assert isinstance(op_schema.args_schema[0], OpStrategy)
+    assert isinstance(op_schema.args_schema[1], OpStrategy)
+    self_strategy: OpStrategy = op_schema.args_schema[0]
+    mesh = self_strategy.mesh
+    src_strategy: OpStrategy = op_schema.args_schema[1]
+    # meta should be the same for all strategies
+    src_meta = src_strategy.strategies[0].output_spec.tensor_meta
+
+    # Following torch broadcasting semantics (https://docs.pytorch.org/docs/stable/notes/broadcasting.html)
+    # - self can not change shape as a result of broadcasting since this is an inplace op
+    # - src can broadcast, but when it does it always does so from the trailing end
+    # e.g. the last dim of 'src' must match up with the last dim of 'self'
     #
-    # For now, I just keep copy working essentially the way it was before this PR,
-    # but split it out so it can be handled separately in the future.
-    num_tensor_args = 2
-    first_input_strategy = op_schema.args_schema[0]
-    assert isinstance(first_input_strategy, OpStrategy)
-    return OpStrategy(
-        [
+    # DTensor semantics for inplace ops also dictates that we may NOT redistribute our 'self' input.
```

Comment:
I don't quite like the fact that we are implementing broadcasting semantic separately, it would be better if you could just reuse the pointwise strategy for `copy_`